examples/ct/ct_handlers.go

```diff
@@ -436,3 +522,86 @@ func marshalAndWriteAddChainResponse(sct ct.SignedCertificateTimestamp, km crypt
 
 	return nil
 }
+
+func parseAndValidateGetEntriesRange(r *http.Request, maxAllowedRange int64) (int64, int64, error) {
+	startIndex, err := strconv.ParseInt(r.FormValue(getEntriesParamStart), 10, 64)
+
+	if err != nil {
+		return 0, 0, err
+	}
+
+	endIndex, err := strconv.ParseInt(r.FormValue(getEntriesParamEnd), 10, 64)
+
+	if err != nil {
+		return 0, 0, err
+	}
+
+	return validateStartAndEnd(startIndex, endIndex, maxAllowedRange)
+}
+
+// validateStartAndEnd applies validation to the range params for get-entries. Returns nil if
+// we accept them or an error giving the reason why not
+func validateStartAndEnd(start, end, maxRange int64) (int64, int64, error) {
+	if start < 0 || end < 0 {
+		return 0, 0, fmt.Errorf("start (%d) and end (%d) parameters must be >= 0", start, end)
+	}
+
+	if start > end {
+		return 0, 0, fmt.Errorf("start (%d) and end (%d) is not a valid range", start, end)
+	}
+
+	numEntries := end - start + 1
+
+	if numEntries > maxRange {
```

Comment:
The FE only rejects ranges that don't make sense like (-10, -8) or (20, 5) or are larger than we're willing to serve. I'll let the backend decide how to handle ranges that exceed the tree size. There are comments about this in the handler.